File: in_Java/0941 Valid Mountain Array.java
class Solution {
    public boolean validMountainArray(int[] A) {
        if(A.length < 3) return false;
        int i = 0;
        int j = A.length-1;
        while(i < A.length-1 && A[i] < A[i+1]){
            i++;
        }
        while(j > 0 && A[j] < A[j-1]){
            j--;
        }
        if(i == 0 || j == A.length-1) return false;
        return i == j;
    }
}

File: in_Python/0941 Valid Mountain Array.py
class Solution(object):
    def validMountainArray(self, A):
        """
        :type A: List[int]
        :rtype: bool
        """
        i,j = 0, len(A)-1
        while i < len(A)-1:
            if A[i+1] > A[i]:
                i += 1
            else:
                break
        while j > 0:
            if A[j-1] > A[j]:
                j -= 1
            else:
                break
        return i != 0 and j != len(A)-1 and i == j